diff --git a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
index 04cadc8..d021e04 100644
--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp
@@ -122,7 +122,7 @@ bool CEGLNativeTypeIMX::CreateNativeDisplay()
   // Force double-buffering
   CEnvironment::setenv("FB_MULTI_BUFFER", "2", 0);
 
-#ifdef HAS_IMXVPU
+#ifdef HAS_IMXVPU_NULL
   // EGL will be rendered on fb0
   m_display = fbGetDisplayByIndex(0);
   m_nativeDisplay = &m_display;
@@ -134,7 +134,7 @@ bool CEGLNativeTypeIMX::CreateNativeDisplay()
 
 bool CEGLNativeTypeIMX::CreateNativeWindow()
 {
-#ifdef HAS_IMXVPU
+#ifdef HAS_IMXVPU_NULL
   m_window = fbCreateWindow(m_display, 0, 0, 0, 0);
   m_nativeWindow = &m_window;
   return true;
@@ -165,7 +165,7 @@ bool CEGLNativeTypeIMX::GetNativeWindow(XBNativeWindowType **nativeWindow) const
 
 bool CEGLNativeTypeIMX::DestroyNativeDisplay()
 {
-#ifdef HAS_IMXVPU
+#ifdef HAS_IMXVPU_NULL
   if (m_display)
     fbDestroyDisplay(m_display);
   m_display =  NULL;
@@ -177,7 +177,7 @@ bool CEGLNativeTypeIMX::DestroyNativeDisplay()
 
 bool CEGLNativeTypeIMX::DestroyNativeWindow()
 {
-#ifdef HAS_IMXVPU
+#ifdef HAS_IMXVPU_NULL
   if (m_window)
     fbDestroyWindow(m_window);
   m_window =  NULL;
